package/gluon-core/luasrc/usr/lib/lua/gluon/wireless.lua

```diff
@@ -2,16 +2,49 @@ local sysconfig = require 'gluon.sysconfig'
 local site = require 'gluon.site'
 local util = require 'gluon.util'
 
+local unistd = require 'posix.unistd'
+local dirent = require 'posix.dirent'
+
 
 local M = {}
 
 local function find_phy_by_path(path)
-	local phy = util.glob('/sys/devices/' .. path .. '/ieee80211/phy*')[1]
-		or util.glob('/sys/devices/platform/' .. path .. '/ieee80211/phy*')[1]
+	local device_path, phy_offset
+	-- Special handling required for multi-phy devices
+	if string.match(path, '+%d$') then
+		device_path, phy_offset = string.match(path, "([^,]+)+(%d)$")
```

Comment:
The literal `+` characters in both match strings should be escaped using `%`.

Reply:
Also, why the `[^,]` pattern? I would use `^(.+)%+(%d+)$`, which also anchors the beginning of the string and handles PHY indices with multiple digits correctly.

Reply:
No specific reason, this regex was tanken from iwinfo source.